source/common/stats/metric_impl.cc

```diff
@@ -15,12 +15,13 @@ MetricHelper::~MetricHelper() {
 }
 
 MetricHelper::MetricHelper(StatName name, absl::string_view tag_extracted_name,
-                           const std::vector<Tag>& tags, SymbolTable& symbol_table) {
+                           const std::vector<Tag>& tags, const StatNameTagVector& stat_name_tags,
+                           SymbolTable& symbol_table) {
   // Encode all the names and tags into transient storage so we can count the
   // required bytes. 2 is added to account for the name and tag_extracted_name,
   // and we multiply the number of tags by 2 to account for the name and value
   // of each tag.
-  const uint32_t num_names = 2 + 2 * tags.size();
+  const uint32_t num_names = 2 + 2 * tags.size() + 2 * stat_name_tags.size();
```

Comment:
Would it expand the PR greatly to remove the `vector<Tag>` arg and change the call-sites to create the StatNameTagVector? If that would be too much, can you leave a TODO?

Reply:
gave it a shot, wasnt too bad so ill go with that